app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,17 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else 
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
+    user.update_attributes(password: params[:password], token: nil) if user
```

Comment:
`update_attributes` is old method for `update`. Also what happens when password would be invalid?